ext/dba/dba_cdb.c

```diff
@@ -145,32 +145,30 @@ DBA_CLOSE_FUNC(cdb)
 DBA_FETCH_FUNC(cdb)
 {
 	CDB_INFO;
+	zend_string *fetched_val = NULL;
 	unsigned int len;
-	char *new_entry = NULL;
 
 #if DBA_CDB_BUILTIN
 	if (cdb->make)
 		return NULL; /* database was opened writeonly */
 #endif
-	if (php_cdb_find(&cdb->c, key, keylen) == 1) {
+	if (php_cdb_find(&cdb->c, ZSTR_VAL(key), ZSTR_LEN(key)) == 1) {
 		while(skip--) {
-			if (php_cdb_findnext(&cdb->c, key, keylen) != 1) {
+			if (php_cdb_findnext(&cdb->c, ZSTR_VAL(key), ZSTR_LEN(key)) != 1) {
 				return NULL;
 			}
 		}
 		len = cdb_datalen(&cdb->c);
-		new_entry = safe_emalloc(len, 1, 1);
+		fetched_val = zend_string_safe_alloc(len-1, 1, 1, /* persistent */ false);
```

Comment:
`zend_string_safe_alloc(len, 1, 0, /* persistent */ false)` would be more natural.

Reply:
Though frankly, I'd just use zend_string_alloc.

Reply:
I used the safe variant as it was using ``safe_emalloc()`` but will change it.